assembly/src/tests.rs

```diff
@@ -105,6 +108,50 @@ fn span_and_simple_if() {
     assert_eq!(expected, format!("{program}"));
 }
 
+// PROGRAM WITH #main CALL
+// ================================================================================================
+#[test]
+fn simple_main_call() {
+    // instantiate assembler
+    let assembler = super::Assembler::default();
+
+    // compile account module
+    let account_path = LibraryPath::new("context::account").unwrap();
+    let account_code = ModuleAst::parse(
+        "\
+    export.account_method_1
+        push.2.1 add
+    end
+    
+    export.account_method_2
+        push.3.1 sub
+    end
+    ",
+    )
+    .unwrap();
+    let account_module = Module::new(account_path, account_code);
+    let _method_roots = assembler
+        .compile_module(
+            &account_module,
+            &mut super::AssemblyContext::new(AssemblyContextType::Module),
+        )
+        .unwrap();
+
+    // compile note 1 program
+    let note_1 =
+        ProgramAst::parse("use.context::account begin call.account::account_method_1 end").unwrap();
+    let _note_1_root = assembler
+        .compile_in_context(note_1, &mut super::AssemblyContext::new(AssemblyContextType::Program))
+        .unwrap();
+
+    // compile note 2 program
+    let note_2 =
+        ProgramAst::parse("use.context::account begin call.account::account_method_2 end").unwrap();
+    let _note_2_root = assembler
+        .compile_in_context(note_2, &mut super::AssemblyContext::new(AssemblyContextType::Program))
+        .unwrap();
```

Comment:
Ideally, we should compile both notes in the same context - but I think it is fine for now.